src/Controller/MainController.java

```diff
@@ -93,63 +93,66 @@ public static void setSpc(StudyPlannerController newSpc) {
 	public static void initialise() {
 		try {
 			ui.showStartup();
-			// If a file is present:
-			if (plannerFile.exists()) {
-				Cipher cipher = Cipher.getInstance("Blowfish");
-				cipher.init(Cipher.DECRYPT_MODE, key64);
-				try (CipherInputStream cis = new CipherInputStream(
+			Cipher cipher = Cipher.getInstance("Blowfish");
+			cipher.init(Cipher.DECRYPT_MODE, key64);
+			try (CipherInputStream cis = new CipherInputStream(
 						new BufferedInputStream(new FileInputStream(plannerFile)), cipher);
-						ObjectInputStream ois = new ObjectInputStream(cis)) {
+					ObjectInputStream ois = new ObjectInputStream(cis)) {
 
+				// If a file is present:
+				if (plannerFile.exists()) {
```

Comment:
What is the benefit of moving this conditional inside the try block?  It looks to me like it will create a situation where the creation of the `BufferedInputStream/FileInputStream` might depend upon a non-existent file.

Reply:
I did all of these because I thought the nested try has to finish with a direct catch statement that is separated from the other try, and that was in the old java versions as you said. I will fix it with a unique solution.